File: csharp/sum-of-multiples/SumOfMultiples.cs
namespace Exercism_sum_of_multiples
{
    using System.Linq;

    public class SumOfMultiples
    {
        public static long To(int[] multipliers, int n)
        {
            return Enumerable.Range(1, n - 1)
                             .Where(e => multipliers.Any(m => e % m == 0))
                             .Sum();
        }
    }
}

File: cpp/sum-of-multiples/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples
{
    long to(std::vector<int> divisors, int n)
    {
        long sum = 0;
        
        for (int i = 1; i < n; i++)
        {
            for (int divisor: divisors)
            {
                if (i % divisor == 0)
                {
                    sum += i;
                    break;
                }
            }
        }
            
        return sum;
    }
}
